### ISSUE TITLE
Rustdoc doesn’t expand macros properly when generating docs

### ISSUE THREAD

**Issue by photino** [rust-lang/rust#28860]
I have defined a macro to bind constants:

```rust
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
pub struct Const(pub i32);

macro_rules! bind_constant {
    ($NAME:ident, $value:expr) => {
        pub const $NAME: Const = Const($value);
    }
}

bind_constant!(A, 10i32);
bind_constant!(B, 20i32);

#[test]
#[should_panic]
fn test_consts() {
    assert_eq!(A, B);
}
```

When I use `rustdoc` to generate docs, it fails to expand the macro properly:
```rust
pub const A: Const = Const($value)
```

**Comment by birkenfeld** [comment]
This was fixed by 24117f3c589dca680bdfe7e193e52e210770dc79 and can be closed.

**Comment by alexcrichton** [comment]
Nice!
